gloss/sys_times.c

```diff
@@ -18,24 +16,19 @@ extern int _gettimeofday(struct timeval *, void *);
    account for user vs system time, but for now we just return the total
    number of cycles since starting the program.  */
 clock_t _times(struct tms *buf) {
-    int rv;
-    // when called for the first time, initialize t0
-    static struct timeval t0;
-    if (t0.tv_sec == 0 && t0.tv_usec == 0)
-        _gettimeofday(&t0, 0);
+    unsigned long long mcc;
+    unsigned long long timebase;
 
-    struct timeval t;
-    _gettimeofday(&t, 0);
+    metal_timer_get_timebase_frequency(0, &timebase);
+    metal_timer_get_cyclecount(0, &mcc);
```

Comment:
We shouldn't assume that we're on hart 0. I'd recommend calling `metal_cpu_get_current_hartid()` so that we get the timebase/cyclecount of the hart we're currently running on.

Reason being is that if we're on a multi-hart design, benchmarks are usually running on hart 1 to make sure we don't get slowed down by accidentally running on a "monitor" hart. Other harts, including hart 0, are usually in a `wfi` loop, which puts them to sleep and I'm pretty sure halts `mcycle`.

Reply:
I added a call to metal_cpu_get_current_hartid, but then realized that the clock scaling could potentially overflow the 64-bit computation. I fixed that with a bit of algebra.